staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -150,16 +152,25 @@ func CheckGeneratedNameError(strategy RESTCreateStrategy, err error, obj runtime
 		return err
 	}
 
-	objectMeta, kind, kerr := objectMetaAndKind(strategy, obj)
+	objectMeta, gvk, kerr := objectMetaAndKind(strategy, obj)
 	if kerr != nil {
 		return kerr
 	}
 
 	if len(objectMeta.GetGenerateName()) == 0 {
+		// If we don't have a generated name, return the original error (AlreadyExists).
+		// When we're here, the user picked a name that is causing a conflict.
 		return err
 	}
 
-	return errors.NewServerTimeoutForKind(kind.GroupKind(), "POST", 0)
+	// If we have a name and generated name, the server picked a name
+	// that already exists.
+	e := errors.NewAlreadyExists(
+		schema.GroupResource{Group: gvk.Group, Resource: gvk.Kind},
```

Comment:
I didn't think you could use gvk.Kind as a Resource?

Reply:
I copied this from other parts of the codebase, they're all using gvk.Kind as GroupResource

Reply:
ah, good catch; Kinds are not resources, people are already extremely confused about this, please get the resource and don't do this (the strategy object should be able to tell you the resource).

Reply:
@lavalamp @ncdc I can't seem to find any way from the available strategy object to retrieve a resource from an object, or a GVK. Is there a method I completely missed?

Reply:
information about the incoming request is in the request info contained in the context (which is not currently plumbed here, but could be)

Reply:
see the `RequestInfoFrom(ctx)` method for pulling that info out of the context

Reply:
Can we modify the function signature here, or are there some compatibility guarantees?

Reply:
passing context here seems reasonable to me

Reply:
Done 👍